runtime/src/main/kotlin/kotlin/native/Text.kt

```diff
@@ -9,45 +9,46 @@ package kotlin.native
 /**
  * Converts an UTF-8 array into a [String]. Replaces invalid input sequences with a default character.
  */
-@Deprecated("Use decodeToString instead", ReplaceWith("decodeToString()"))
 public fun ByteArray.stringFromUtf8() : String {
-    @Suppress("DEPRECATION")
     return this.stringFromUtf8(0, this.size)
 }
 
 /**
  * Converts an UTF-8 array into a [String]. Replaces invalid input sequences with a default character.
  */
-@Deprecated("Use decodeToString instead", ReplaceWith("decodeToString(start, start + size)"))
 public fun ByteArray.stringFromUtf8(start: Int = 0, size: Int = this.size) : String {
     checkBoundsIndexes(start, start + size, this.size)
-    return stringFromUtf8Impl(start, size)
+    return stringFromUtf8Impl(start, realSize(this, size))
 }
 
 @SymbolName("Kotlin_ByteArray_stringFromUtf8")
 internal external fun ByteArray.stringFromUtf8Impl(start: Int, size: Int) : String
 
+private fun realSize(byteArray: ByteArray, size: Int): Int {
+    var realSize = 0
+    while (realSize < size && byteArray[realSize] != 0.toByte()) {
+        realSize++
+    }
+    return realSize
+}
+
 /**
  * Converts an UTF-8 array into a [String].
  * @throws [IllegalCharacterConversionException] if the input is invalid.
  */
-@Deprecated("Use decodeToString instead", ReplaceWith("decodeToString(throwOnInvalidSequence = true)"))
 public fun ByteArray.stringFromUtf8OrThrow() : String {
-    @Suppress("DEPRECATION")
     return this.stringFromUtf8OrThrow(0, this.size)
 }
 
 /**
  * Converts an UTF-8 array into a [String].
  * @throws [IllegalCharacterConversionException] if the input is invalid.
  */
-@Deprecated("Use decodeToString instead", ReplaceWith("decodeToString(start, start + size, throwOnInvalidSequence = true)"))
 public fun ByteArray.stringFromUtf8OrThrow(start: Int = 0, size: Int = this.size) : String {
     checkBoundsIndexes(start, start + size, this.size)
     try {
-        return stringFromUtf8OrThrowImpl(start, size)
+        return stringFromUtf8OrThrowImpl(start, realSize(this, size))
     } catch (e: CharacterCodingException) {
-        @Suppress("DEPRECATION")
         throw IllegalCharacterConversionException()
```

Comment:
Now that `ByteArray.stringFromUtf8OrThrow()` is not deprecated, `IllegalCharacterConversionException` is also not deprecated. Because throwing deprecated exception from non-deprecated function looks odd.

Reply:
> > @ilya-g If you are asking about zero-terminated strings: `./gradlew interop_zlib`
> 
> I do not see where it fails on the build server before this change. @olonho Could you clarify?

It doesn't fail as it needs platform libraries, and is not being run by default.